### PR TITLE
(Buf)AmpGate RST & SC examples

### PR DESCRIPTION
**Pull request by tedmoore**

### CODE REVIEW

File: doc/AmpGate.rst [hunk 1]
```diff
@@ -1,16 +1,16 @@
-:digest: Amplitude-based Gating Slicer
+:digest: Gate Detection on a Signal
 :species: slicer
 :sc-categories: Libraries>FluidDecomposition
 :sc-related: Guides/FluidCorpusManipulationToolkit
 :see-also: BufAmpGate, AmpSlice, OnsetSlice, NoveltySlice, TransientSlice
-:description: This class implements an amplitude-based slicer, with various customisable options and conditions to detect absolute amplitude changes as onsets and offsets.
-:discussion: 
-   FluidAmpSlice is based on an envelop follower on a highpassed version of the signal, which is then going through a Schmidt trigger and state-aware time contraints. The example code below is unfolding the various possibilites in order of complexity.
+:description: Absolute amplitude threshold gate detector on a real-time signal
```
Comment (jamesb93) [inline doc/AmpGate.rst:6]: I think "realtime" as opposed to "real-time" is the more consistent way to put it with the rest of the documentation.

Comment (jamesb93) [reply doc/AmpGate.rst:6]: It also makes it more consistent when the inevitable "non-realtime" appears

**Review by jamesb93 — CHANGES_REQUESTED** [review]
I think this missing a didactic example about lookback/ahead which the Max documentation seems to have. I think it's valuable and already exists in Max, though I'm not sure about SC and maybe you removed it or it was never there in the first place.


image

**Comment by tremblap** [comment]
> I think this missing a didactic example about lookback/ahead which the Max documentation seems to have. 

it definitely was in there originally but everything was plotting :)

File: doc/AmpGate.rst (hunk 1 above)
Comment (tremblap) [reply doc/AmpGate.rst:6]: agreed

**Review by tremblap — CHANGES_REQUESTED** [review]
I also miss (like @jamesb93 ) the visual example that were provided as tutorial to understand it all.

**Comment by tremblap** [comment]
obviously the buf version with waveform to look at is much clearer!

**Comment by tedmoore** [comment]
> the visual example that were provided as tutorial to understand it all.

I'm not sure what you're referring to. I just looked back at the help files. If you're referring to something that is in here (see picture below). I deleted all this--it was testing the parameters one by one and felt way cold and uninviting. If there's a particular visual display that you want added back in, I can make that happen.

Screenshot 2022-05-31 at 15 39 29

**Comment by tremblap** [comment]
> I'm not sure what you're referring to.

that code. in max it is less ugly because of scope and rt but it helps - see the picture @jamesb93 showed above.

**Comment by tedmoore** [comment]
I would prefer to not have code in the help file that looks like this--it's not clear what a user is supposed to take a way from it or how to use it. I think the Max version is more clear and approachable. Can you specify for me in a few sentences what is most important to elucidate or demonstrate for the user and I can try to do that succinctly?

**Comment by tremblap** [comment]
I think the Max version's narrative (changing setting, described expected behaviour, look at the gate) is quite good. I translated it from the SC example initially, so I think it is doing that, but if you think of a better, more interactive way to look at gates with commented dynamic preset changes, then I'm happy. The idea is that one can observe the wave and the gate, and hear the state change. it is very multimodal teaching in max...

**Comment by jamesb93** [comment]
> I think the Max version's narrative (changing setting, described expected behaviour, look at the gate) is quite good. I translated it from the SC example initially, so I think it is doing that, but if you think of a better, more interactive way to look at gates with commented dynamic preset changes, then I'm happy. The idea is that one can observe the wave and the gate, and hear the state change. it is very multimodal teaching in max...

Let's remember that this is up for debate before Max get's changed also. I don't love the narrative style – it has the potential to be confusing when it is as long and didactic as this. I actually don't currently think the Max one offers that much either, sorry. Some vertical markers move around a synthetic waveform. As @tedmoore said it is difficult for a user to take much away from this without already knowing what they need to know. I prefer something that is like an actual narrative based on a problematic use case, like the Constanzo drum sample and capturing different levels of intricacy.

**Comment by jamesb93** [comment]
> Ok, well, I'm glad @jamesb93 said so because I agree with him. I think tweaking the parameters needs some motivation and some highlighting like, "see, now the slice point is specified before the drum hit onset, even though it's triggered after the onset" and being zoomed into to see the few hundred samples difference. I could poke at a narrative of trying to see these parameters move around a drum hit perhaps?

I think that is a good path forward – contextualise this nitty gritty in real use case so it sticks.

**Pull request merged**